comparison/comparison.go

```diff
@@ -0,0 +1,509 @@
+	var err error
+	var status coordinator.Status
+	mlog.Info("starting bounded load-test")
+	if err := t.StartCoordinator(coordConfig); err != nil {
+		return status, err
+	}
+
+	var canceled bool
+	select {
+	case <-cancelCh:
+		mlog.Info("cancelling load-test")
+		canceled = true
+	case <-time.After(d):
+	}
+
+	mlog.Info("stopping bounded load-test")
+	status, err = t.StopCoordinator()
+	if err != nil {
+		return status, err
+	}
+
+	if canceled {
+		return status, errors.New("canceled")
+	}
+
+	// TODO: remove this once MM-30326 has been merged and a new release
+	// published.
+	status.StopTime = time.Now()
+
+	return status, nil
+}
+
+func runUnboundedLoadTest(t *terraform.Terraform, coordConfig *coordinator.Config, cancelCh <-chan struct{}) (coordinator.Status, error) {
+	mlog.Info("starting unbounded load-test")
+	if err := t.StartCoordinator(coordConfig); err != nil {
+		return coordinator.Status{}, err
+	}
+
+	for {
+		status, err := t.GetCoordinatorStatus()
+		if err != nil {
+			return status, err
+		}
+
+		if status.State == coordinator.Done {
+			mlog.Info("unbounded load-test has completed")
+			return status, nil
+		}
+
+		if status.State != coordinator.Running {
+			return status, errors.New("coordinator should be running")
+		}
+
+		select {
+		case <-cancelCh:
+			mlog.Info("cancelling load-test")
+			if status, err := t.StopCoordinator(); err != nil {
+				return status, err
+			}
+			return coordinator.Status{}, errors.New("canceled")
+		case <-time.After(time.Minute):
+		}
+	}
+}
+
+func initLoadTest(t *terraform.Terraform, config *deployment.Config, buildCfg BuildConfig, cancelCh <-chan struct{}) error {
+	output, err := t.Output()
+	if err != nil {
+		return err
+	}
+
+	if !output.HasAppServers() {
+		return errors.New("no app servers in this deployment")
+	}
+
+	extAgent, err := ssh.NewAgent()
+	if err != nil {
+		return err
+	}
+
+	agentClient, err := extAgent.NewClient(output.Agents[0].PublicIP)
+	if err != nil {
+		return fmt.Errorf("error in getting ssh connection %w", err)
+	}
+	defer agentClient.Close()
+
+	appClients := make([]*ssh.Client, len(output.Instances))
+	for i, instance := range output.Instances {
+		client, err := extAgent.NewClient(instance.PublicIP)
+		if err != nil {
+			return fmt.Errorf("error in getting ssh connection %w", err)
+		}
+		defer client.Close()
+		appClients[i] = client
+	}
+
+	type cmd struct {
+		msg     string
+		value   string
+		clients []*ssh.Client
+	}
+
+	stopCmd := cmd{
+		msg:     "Stopping app servers",
+		value:   "sudo systemctl stop mattermost",
+		clients: appClients,
+	}
+
+	buildFileName := filepath.Base(buildCfg.URL)
+	installCmd := cmd{
+		msg:     "Installing app",
+		value:   fmt.Sprintf("tar xzf %s && cp /opt/mattermost/config/config.json . && sudo rm -rf /opt/mattermost && sudo mv mattermost /opt/ && mv config.json /opt/mattermost/config/", buildFileName),
```

Comment:
Maybe good to do `cd /home/ubuntu` before this? We don't know what the default `$HOME` is